web/client-api/src/main/java/io/deephaven/web/client/api/storage/JsStorageService.java

```diff
@@ -0,0 +1,175 @@
+        return Callbacks.<FetchFileResponse, Object>grpcUnaryPromise(c -> client().fetchFile(req, metadata(), c::apply))
+                .then(response -> {
+                    if (response.hasEtag() && response.getEtag().equals(etag)) {
+                        return Promise.resolve(new FileContents(etag));
+                    }
+                    Blob contents = new Blob(JsArray.of(
+                            Blob.ConstructorBlobPartsArrayUnionType.of(response.getContents_asU8().slice().buffer)));
+                    return Promise.resolve(new FileContents(contents, response.getEtag()));
+                });
+    }
+
+    /**
+     * Deletes the item at the given path. Directories must be empty to be deleted.
+     *
+     * @param path the path of the item to delete
+     * @return a promise with no value on success, or an error.
+     */
+    @JsMethod
+    public Promise<Void> deleteItem(String path) {
+        DeleteItemRequest req = new DeleteItemRequest();
+        req.setPath(path);
+        return Callbacks
+                .<DeleteItemResponse, Object>grpcUnaryPromise(c -> client().deleteItem(req, metadata(), c::apply))
+                .then(response -> Promise.resolve((Void) null));
+    }
+
+    /**
+     * Saves the provided contents to the given path, creating a file or replacing an existing one. The optional newFile
+     * parameter can be passed to indicate that an existing file must not be overwritten, only a new file created.
+     *
+     * Note that directories must be empty to be overwritten.
+     *
+     * @param path the path of the file to write
+     * @param contents the contents to write to that path
+     * @param newFile true to force a new file to be created, false to allow an existing file to be overwritten
+     * @return a promise with no value on success, or an error.
+     */
+    @JsMethod
+    public Promise<Void> saveFile(String path, FileContents contents, @JsOptional Boolean newFile) {
+        return contents.arrayBuffer().then(ab -> {
+            SaveFileRequest req = new SaveFileRequest();
+            req.setContents(new Uint8Array(ab));
+            req.setPath(path);
+
+            if (newFile != null) {
+                req.setNewFile(newFile);
+            }
+
+            return Callbacks
+                    .<SaveFileResponse, Object>grpcUnaryPromise(c -> client().saveFile(req, metadata(), c::apply))
+                    .then(response -> Promise.resolve((Void) null));
+        });
+    }
+
+    /**
+     * Moves (and/or renames) an item from its old path to its new path. The optional newFile parameter can be passed to
+     * enforce that an existing item must not be overwritten.
+     *
+     * Note that directories must be empty to be overwritten.
+     *
+     * @param oldPath the path of the existing item
+     * @param newPath the new path to move the item to
+     * @param newFile true to forbid an existing file be created, false to allow an existing file to be overwritten
```

Comment:
This comment wording doesn't make sense. This param/comment should match the `saveFile` method as well, though I don't think `newFile` is the best name. I'd probably change it to `allowOverwrite`, `disallowOverwrite`, or something like that instead.

Reply:
To avoid confusion from double negatives (disallowOverwrite=false), I would need to make this "default to safe" (since proto defaults to false, so allowOverwrite=false would mean that callers would need to specify that they want to overwrite files in both contexts.